### ISSUE TITLE
ETCD has not been building artefacts for darwin/arm64 to support Apple Silicon

### ISSUE THREAD

**Issue by camilamacedo86** [etcd-io/etcd#14001]
### What happened?

We would like to be able to download binaries for the architecture darwin/arm64:

```
curl -sLO https://github.com/coreos/etcd/releases/download/${ETCD_VERSION}/${ETCD_BASE_NAME}.zip
```

Such as `etcd-v3.5.1-darwin-arm64.zip.`

### What did you expect to happen?

ETCD produces artefacts for this platform and architecture so that we can begin to consume it.

### How can we reproduce it (as minimally and precisely as possible)?

curl -sLO https://github.com/coreos/etcd/releases/download/3.5.1/etcd-v3.5.1-darwin-arm64.zip

**Comment by camilamacedo86** [comment]
I can see that the arm64 for mac os was added.
So, we would need to have a new build and then it would be solved:
https://github.com/etcd-io/etcd/blob/main/.github/workflows/build.yaml#L13

Could we do the release?

**Comment by camilamacedo86** [comment]
@ serathius, @ahrtr could you help out?

**Comment by ahrtr** [comment]
I don't think `ARM64 + Darwin` is supported. FYI [Current support](https://etcd.io/docs/v3.5/op-guide/supported-platform/#current-support)

cc @serathius @ptabor @spzala

**Comment by camilamacedo86** [comment]
Hi @ahrtr ,

The goal of this issue is to ask for the darwin/arm64 bin to begin to be built.
Also, I can see that it was added here : https://github.com/etcd-io/etcd/blob/main/.github/workflows/build.yaml#L12-L13

So, probably we are missing a new release?

**Comment by spzala** [comment]
@camilamacedo86 here is a related discussion on initial projected support, https://github.com/etcd-io/etcd/pull/13545 Our previous release managers are not so active lately, and @serathius is working on simplifying and improving the process. I believe the future release assets should have binaries that you asking for, but I will wait on him for a better answer. Thanks!

**Comment by camilamacedo86** [comment]
Hi @spzala, 

Thank you. We had a release 17 days ago: https://github.com/etcd-io/etcd/releases/tag/v3.5.4
So, it seems that we are still missing something to have this bin or we just do not add to the release the arm64 bin.

**Comment by camilamacedo86** [comment]
Hi @spzala, 

Any change on this one?
Do you think that we could have this bin?

**Comment by protosam** [comment]
Building 3.5.4 seems to work on darwin-arm64 for me. The `build.yaml` could just create/update releases when tags are pushed. Depending on which action is used, there is more or less flexibility. The one in my example seems good enough if you want to automate filling in release details.

Example build.yaml:
https://github.com/UniversalShipping/etcd/blob/730f3ec0c2be23838f8a5bf207d1b4d977039cfa/.github/workflows/build.yaml

Results: https://github.com/UniversalShipping/etcd/releases


@serathius / @hexfusion would be desirable over manually pushing these?

**Comment by camilamacedo86** [comment]
@protosam that would be very helpful. Could you please add it to the releases page so that we can download it with `curl -sLO https://github.com/coreos/etcd/releases/download/${ETCD_VERSION}/${ETCD_BASE_NAME}.zip` from 3.5.4 ?

 
`

**Comment by serathius** [comment]
@protosam Automating etcd releases is a great idea, however it's a little bit complicated. Etcd release process has additional steps over just building images, for example all binaries need to be signed by on one of the maintainers. Instead of `scripts/build.sh` we run `scripts/release.sh`.

I'm not saying that we shouldn't do it, it's just more effort. Work is tracked in https://github.com/etcd-io/etcd/issues/13980, please let me know if you want to help.

**Comment by serathius** [comment]
@camilamacedo86 I will be looking into making sure that darwin/arm64 is included in next v3.5.5 release expected within weeks.

**Comment by serathius** [comment]
Due to lack of testing we dropped official support for ARM64. https://etcd.io/docs/v3.6/op-guide/supported-platform/

I can send a PR that would backport https://github.com/etcd-io/etcd/pull/13545 to release-3.5, however want to make sure we are consistent with other platforms. Technically we still build and publish images for windows-amd64 and darvin-amd64, so it's not unusual. 

What's your opinion @ptabor @ahrtr @spzala ?

**Comment by spzala** [comment]
@serathius thanks, sounds good to backport and continue support as tier 3.

**Comment by ahrtr** [comment]
Agreed, it's OK to support it as tier 3, since it's clearly requested by some people. cc @dims

**Comment by dims** [comment]
thanks @serathius @spzala @ahrtr

**Comment by dims** [comment]
thanks a ton @serathius !

**Comment by wuhuizuo** [comment]
Will it back port to release 3.4?

**Comment by jmhbnz** [comment]
> Will it back port to release 3.4?

Hey @wuhuizuo, thanks for your question.  The short answer is no.

The etcd project has tier 1 support for `arm64` from `release-3.5` onward ([refer announcement](https://groups.google.com/g/etcd-dev/c/s3srSmDN92k/m/Hb_lyL9ZCAAJ)). 

To the best of my knowledge it is not a priority for the project to try to get `release-3.4` to tier 1 support for `arm64`. We have very little capacity currently and must stay focused on our [roadmap priorities](https://github.com/etcd-io/etcd/blob/main/Documentation/contributor-guide/roadmap.md#v360).
